state/executor.go

```diff
@@ -198,11 +188,12 @@ func (t *Transition) Write(txn *types.Transaction) error {
 	// Make a local copy and apply the transaction
 	msg := txn.Copy()
 
-	gasUsed, failed, err := t.Apply(msg)
+	result, err := t.Apply(msg)
 	if err != nil {
 		fmt.Printf("Apply err: %v", err)
+		return err
```

Comment:
@Vuksan the `err` that we are checking against here is a violation of a consensus rule, and not a VM error which we check with `ExecutionResult.Failed()`. Because of that, the presence of the `err` value should not end up creating a receipt as it broke the consensus rules.

Reply:
You're completely right. 💯